### PR TITLE
New metric schema

### PR DESCRIPTION
**Pull request by tiffanyfj**

### CODE REVIEW

File: pcm/pcm.go [hunk 1]
```diff
@@ -64,17 +65,15 @@ func (p *PCM) Data() map[string]interface{} {
 }
 
 // CollectMetrics returns metrics from pcm
-func (p *PCM) CollectMetrics(mts []plugin.PluginMetricType) ([]plugin.PluginMetricType, error) {
-	metrics := make([]plugin.PluginMetricType, len(mts))
+func (p *PCM) CollectMetrics(mts []plugin.MetricType) ([]plugin.MetricType, error) {
+	metrics := make([]plugin.MetricType, len(mts))
```
Comment (marcin-krolik) [inline pcm/pcm.go]: Consider populating of incoming ```mts []pluging.MetricType``` with metric values instead creating new instance and returning it. This way all tags, description and units are preserved.

File: pcm/pcm.go [hunk 2]
```diff
@@ -84,12 +83,12 @@ func (p *PCM) CollectMetrics(mts []plugin.PluginMetricType) ([]plugin.PluginMetr
...
 // GetMetricTypes returns the metric types exposed by pcm
-func (p *PCM) GetMetricTypes(_ plugin.PluginConfigType) ([]plugin.PluginMetricType, error) {
-	mts := make([]plugin.PluginMetricType, len(p.keys))
+func (p *PCM) GetMetricTypes(_ plugin.ConfigType) ([]plugin.MetricType, error) {
+	mts := make([]plugin.MetricType, len(p.keys))
 	p.mutex.RLock()
 	defer p.mutex.RUnlock()
 	for i, k := range p.keys {
-		mts[i] = plugin.PluginMetricType{Namespace_: strings.Split(strings.TrimPrefix(k, "/"), "/")}
+		mts[i] = plugin.MetricType{Namespace_: core.NewNamespace(strings.Split(strings.TrimPrefix(k, "/"), "/")...)}
```
Comment (marcin-krolik) [inline pcm/pcm.go]: Would be good to provide also ```plugin.MetricType.Description_``` and ```plugin.MetricType.Units_``` if possible

Comment (tiffanyfj) [inline pcm/pcm.go]: Great idea--will do that as a separate PR. Thanks.

**Comment by jcooklin** [comment]
LGTM
